inc/scroom/layeroperations.hh

```diff
@@ -142,3 +142,18 @@ public:
   virtual void reduce(Tile::Ptr target, const ConstTile::Ptr source, int x, int y);
 };
 
+class OperationsCMYK : public CommonOperations
+{
+public:
+  static Ptr create(int bps);
+  OperationsCMYK(uint16_t bps);
+  virtual ~OperationsCMYK()
+  {}
+
+  virtual int getBpp();
+  virtual Scroom::Utils::Stuff cache(const ConstTile::Ptr tile);
+  virtual void reduce(Tile::Ptr target, const ConstTile::Ptr source, int x, int y);
+private:
+  uint16_t bps;
```

Comment:
Why is this a `uint16_t`? Your default type should be (`unsigned`) `int`. Anything else should have a good reason. To me, this code suggests you consider supporting `bps` values up to 64000 or something?

Reply:
To me, using an `unsigned int` in this case suggests that we're planning to support up to 2147483647 bits per sample. Additionally, the TIFF specification defines the number of bits per sample to be a 16-bit unsigned integer. As such, I think it's only natural to use a `uint16_t`.

Reply:
I think I would be in favor of that